mods/default/functions.lua

```diff
@@ -757,3 +757,11 @@ function default.can_interact_with_node(player, pos)
 
 	return false
 end
+
+function default.log_action(player, pos, message)
+	-- only log actions of real players
+	if player and not player.is_fake_player and player:is_player() then
```

Comment:
After some more consideration, I'm not quite happy with this (particularly if it's not documented). It relies on the `player.is_fake_player` field which is neither a MTE nor a MTG API and rather a convention a couple mods have stuck to. I wouldn't want MTG to add compatibility for random mods this way; rather, those mods should be adding compat for MTG, or where inconvenient, a decent MTG API should be created; designing the API to comply with mods seems odd to me though.

Reply:
without `player.is_fake_player` we can not see if it is a real player. 
hopper and pipeworks mod set, for example, `is_player = function() return true end`:
- https://github.com/minetest-mods/hopper/blob/master/utility.lua#L61
- https://github.com/mt-mods/pipeworks/blob/master/common.lua#L183

This PR would then no longer make sense.

Reply:
the `if player` and `player:is_player()` still appear redundant to me: obviously `player:is_player()` is supposed to be a tautology - this callback does, as it's currently documented, not accept `nil` or invalid players and it's not on MTG to change that

Reply:
fixed in https://github.com/minetest/minetest_game/pull/2941/commits/eb162e6c2951c9e76f38b8157b40c3665fb5334a